src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -674,97 +548,140 @@ private static LoadDocumentEntity GetDocumentPages(PostedDataEntity postedData,
             // get/set parameters
             string documentGuid = postedData.guid;
             string password = string.IsNullOrEmpty(postedData.password) ? null : postedData.password;
-            if (!File.Exists(documentGuid)) 
-            { 
-                throw new GroupDocsViewerException("File not found."); 
+
+            var fileFolderName = Path.GetFileName(documentGuid).Replace(".", "_");
+            string fileCacheSubFolder = Path.Combine(cachePath, fileFolderName);
+            string htmlResourcePrefix = "/viewer/resources/";
+
+            if (!File.Exists(documentGuid))
+            {
+                throw new GroupDocsViewerException("File not found.");
             }
 
-            string cachePath;
-            ViewerSettings settings = GetViewerSettings(documentGuid, out cachePath);
+            IViewerCache cache = new FileViewerCache(cachePath, fileCacheSubFolder);
 
-            using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, GetLoadOptions(password), settings))
+            LoadDocumentEntity loadDocumentEntity = new LoadDocumentEntity();
```

Comment:
Codacy found an issue: [Remove this useless assignment to local variable 'loadDocumentEntity'.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933705)